### PR TITLE
Show LibXML errors in Diagnostics Dialog

### PR DESCRIPTION
**Pull request by ybznek**
Add LibXML errors to diagnostics dialog according to
https://github.com/OpenSCAP/scap-workbench/issues/27

### CODE REVIEW

**Comment by ybznek** [comment]
Still not ready to Merge. It need some more "beautiful" output (and don't destroy xml tags in html output)

screenshot_2015-08-26_16-11-12

File: src/LibXmlErrorGuard.cpp [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+bool LibXmlErrorGuard::isEmpty()
+{
+    return errorMessage.isEmpty();
+}
+
+void LibXmlErrorGuard::libxmlErrorCallback(QString* message, const char* format, ... )
+{
+    try
+    {
+      va_list args;
```
Comment (ybznek) [inline src/LibXmlErrorGuard.cpp]: Uff, bad indentation again...

File: src/LibXmlErrorGuard.cpp [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    // set default libxml error handler
+    xmlSetGenericErrorFunc(NULL, (xmlGenericErrorFunc)NULL);
+}
+
+const QString& LibXmlErrorGuard::getMessage() const
+{
+   return errorMessage;
+}
+
+bool LibXmlErrorGuard::isEmpty()
```
Comment (mpreisler) [inline src/LibXmlErrorGuard.cpp]: const

File: src/LibXmlErrorGuard.cpp [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    {
+        va_list args;
+        va_start(args, format);
+        QString newMessage = QString().vsprintf(format, args);
+        va_end(args);
+        message->append(newMessage);
+    }
+    catch(...)
+    {
+        ; // Possible exceptions aren't safe in C callback
```
Comment (mpreisler) [inline src/LibXmlErrorGuard.cpp]: I don't see how exceptions can happen in the block, maybe I am missing something. Qt doesn't throw.

Comment (ybznek) [inline src/LibXmlErrorGuard.cpp]: I think "new" somewhere inside of these functions can throw exception, or cannot?

Comment (mpreisler) [inline src/LibXmlErrorGuard.cpp]: It can throw `std::bad_alloc` AFAIK but if that happens all is lost anyway :-)

**Comment by ybznek** [comment]
screenshot_2015-08-26_18-20-43

Any suggestions?

**Comment by mpreisler** [comment]
Looks great.

**Comment by mpreisler** [comment]
Discussed this with @ybznek , the fix has to go into OpenSCAP and not SCAP Workbench.

**Comment by ybznek** [comment]
Probably... It will need some fix in Workbench too (printing preformated xml error message/escape xml before output/..) But I can cherry-pick these changes...

**Pull request closed without merging**